src/feature/control/control_events.c

```diff
@@ -1774,31 +1775,25 @@ control_event_guard(const char *nickname, const char *digest,
  * a smartlist_t containing (key, value, ...) pairs in sequence.
  * <b>value</b> can be NULL. */
 int
-control_event_conf_changed(const smartlist_t *elements)
+control_event_conf_changed(config_line_t *elements)
 {
-  int i;
   char *result;
   smartlist_t *lines;
-  if (!EVENT_IS_INTERESTING(EVENT_CONF_CHANGED) ||
-      smartlist_len(elements) == 0) {
+  if (!elements) {
     return 0;
   }
   lines = smartlist_new();
-  for (i = 0; i < smartlist_len(elements); i += 2) {
-    char *k = smartlist_get(elements, i);
-    char *v = smartlist_get(elements, i+1);
-    if (v == NULL) {
-      smartlist_add_asprintf(lines, "650-%s", k);
+  for (config_line_t *line = elements; line; line = line->next) {
```

Comment:
Note if you make the parameter type a pointer to const, `line` also needs a const qualifier.